drv/lpc55-sprot-server/src/handler.rs

```diff
@@ -511,6 +571,55 @@ impl<'a> Handler {
                 };
                 Ok((RspBody::Attest(rsp), None))
             }
+            ReqBody::Attest(AttestReq::TqCert {
+                index,
+                offset,
+                size,
+            }) => {
+                // This command returns a variable amount of data that belongs
+                // in the trailing data region of the response. We return a
+                // marker struct with the data necessary retrieve this data so
+                // the work can be done elsewhere.
+                Ok((
+                    RspBody::Attest(Ok(AttestRsp::TqCert)),
+                    Some(TrailingData::AttestTqCert {
+                        index,
+                        offset,
+                        size,
+                    }),
+                ))
+            }
+            ReqBody::Attest(AttestReq::TqSign {
+                write_size,
+            }) => Ok((
+                RspBody::Attest(Ok(AttestRsp::TqSign)),
+                Some(TrailingData::AttestTqSign {
+                    hash: &req.blob,
+                    write_size,
+                }),
+            )),
+            ReqBody::Attest(AttestReq::TqSignLen) => {
+                let rsp = match self.attest.tq_sign_len() {
+                    Ok(l) => Ok(AttestRsp::TqSignLen(l)),
```

Comment:
**suggestion:** This could be just `self.attest.tq_sign_len().map(AttestResp::TqSignLen)` I believe.

And similarly below.